src/HandlerV1.sol

```diff
@@ -44,13 +46,27 @@ contract HandlerV1 is IHandler, Context {
         host.storeConsensusUpdateTime(host.timestamp());
 
         uint256 commitmentsLen = intermediates.length;
+        uint256 latestHeight = host.latestStateMachineHeight();
+        uint256 stateMachineId = host.stateMachineId();
         for (uint256 i = 0; i < commitmentsLen; i++) {
             IntermediateState memory intermediate = intermediates[i];
+            if (intermediate.stateMachineId != stateMachineId) {
```

Comment:
On second thought, we should only return a single intermediate state, the latest one.